x/pairing/keeper/msg_server_stake_provider_test.go

```diff
@@ -812,3 +813,47 @@ func TestUnfreezeWithDelegations(t *testing.T) {
 	_, err = ts.TxPairingUnfreezeProvider(provider, ts.spec.Index)
 	require.NoError(t, err)
 }
+
+func TestCommisionChange(t *testing.T) {
```

Comment:
add a comment that sunnarizes all the scenarios you're testing (you can also add comments inside the function, some of the checks are unexplained)